### PR TITLE
chore: minor typo cleanup

### PR DESCRIPTION
**Pull request by nschonni**
## Motivation

Took a quick pass with cSpell to fix some typos. Avoided some of the intentional ones left for backward compat

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

N/A

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: website/docs/guides/docs/versioning.md [hunk 1]
```diff
@@ -259,7 +259,7 @@ The [@hello](hello.md#paginate) document is great!
 See the [Tutorial](../getting-started/tutorial.md) for more info.
 ```
 
-### Global or versioned colocated assets {#global-or-versioned-colocated-assets}
+### Global or versioned collocated assets {#global-or-versioned-collocated-assets}
```
Comment (Josh-Cena) [inline website/docs/guides/docs/versioning.md:262]: "collocate" is different from "co-locate", I guess? Maybe add a hyphen here instead?

Comment (nschonni) [reply website/docs/guides/docs/versioning.md:262]: https://www.merriam-webster.com/dictionary/collocate seems to be correct

Comment (Josh-Cena) [reply website/docs/guides/docs/versioning.md:262]: Okay that makes sense

**Comment by nschonni** [comment]
Looks like a snapshot might need to be updated. Trying to rebase and try that now

**Comment by nschonni** [comment]
Looks like you beat me to it @Josh-Cena

**Comment by nschonni** [comment]
Let me know if you want me to force push as the single commit

**Comment by Josh-Cena** [comment]
Haha I think I've fixed everything :D

**Comment by Josh-Cena** [comment]
Just waiting for the tests to all pass. Unfortunately I don't think we enabled auto-merge

**Comment by Josh-Cena** [comment]
We only squash-merge so no need to force-push yourself. Thank you very much @nschonni! Wanted to cSpell our repo for a long time.

**Comment by nschonni** [comment]
no problem, I didn't take the time to really build up a dictionary to add it to the CI because there were still thousands of things flagged for now, but it's mostly just words for packages and some i18n stuff

**Comment by Josh-Cena** [comment]
Yeah, I know... I use the vscode extension and my workspace dictionary is bloating up :D If you want to integrate it into the CI, PRs are welcome! I can help you complete it.

**Pull request merged**